ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java

```diff
@@ -162,30 +162,50 @@ public void onPong(Buffer buffer) {
 
       @Override
       public void onMessage(ResponseBody response) throws IOException {
-        String message;
-        try {
-          if (response.contentType() == WebSocket.BINARY) {
-            message = Base64.encodeToString(response.source().readByteArray(), Base64.NO_WRAP);
-          } else {
-            message = response.source().readUtf8();
+        WritableMap params = Arguments.createMap();
+        params.putInt("id", id);
+
+        if (mBlobsEnabled.containsKey(id) &&
+                mBlobsEnabled.get(id) &&
+                response.contentType() == WebSocket.BINARY) {
+          byte[] data;
+          try {
+            data = response.source().readByteArray();
```

Comment:
I'm wondering if this can cause OOMs quite easily if the data is large. But the blob does have to be copied and held in memory probably anyway..